.github/workflows/modernisation-platform.yml

```diff
@@ -0,0 +1,70 @@
+---
+name: modernisation-platform
+on:
+  push:
+    branches:
+      - main
+      - feature/*
+    paths:
+      - 'modernisation-platform/**'
+      - '.github/workflows/modernisation-platform.yml'
+  pull_request:
+    branches:
+      - main
+    types: [opened, edited, reopened, synchronize]
+    paths:
+      - 'modernisation-platform/**'
+      - '.github/workflows/modernisation-platform.yml'
+  workflow_dispatch:
+env:
+  AWS_ACCESS_KEY_ID:  ${{ secrets.AWS_ACCESS_KEY_ID }}
+  AWS_SECRET_ACCESS_KEY:  ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+  TF_IN_AUTOMATION: true
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  plan-modernisation-platform:
+    name: Plan - modernisation-platform pipelines
+    runs-on: ubuntu-latest
+    if: github.ref != 'refs/heads/main' || github.event_name == 'workflow_dispatch'
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2.3.4
+      - name: Load and Configure Terraform
+        uses: hashicorp/setup-terraform@v1.3.2
+        with:
+          terraform_version: 1.0.1
+          terraform_wrapper: false
+      - name: Terraform plan - modernisation-platform pipelines
+        run: |
+          echo "Terraform plan - ${TF_ENV}"
+          bash scripts/terraform-init.sh modernisation-platform
+          terraform -chdir="modernisation-platform" workspace select "core-shared-services-${TF_ENV}"
+          bash scripts/terraform-plan.sh modernisation-platform
+        env:
+          TF_ENV: production
+
+  deploy-modernisation-platform-:
+    name: Deploy - modernisation-platform pipelines
+    runs-on: ubuntu-latest
+    if: github.ref != 'refs/heads/main' || github.event_name == 'workflow_dispatch'
```

Comment:
is it time to set github.ref == 'refs/head/main'?

Reply:
Talked this through on slack. For Image builder change processes, deploying from branch does make some sense. Suggest adding this to the README as a suggested convention.